connector/grafanacloudconnector/connector.go

```diff
@@ -0,0 +1,117 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package grafanacloudconnector // import "github.com/open-telemetry/opentelemetry-collector-contrib/connector/grafanacloudconnector"
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+const (
+	hostInfoMetric     = "traces_host_info"
+	hostIdentifierAttr = "grafana.host.id"
+)
+
+var _ connector.Traces = (*connectorImp)(nil)
+
+type connectorImp struct {
+	config Config
+	logger *zap.Logger
+
+	started      bool
+	done         chan struct{}
+	shutdownOnce sync.Once
+
+	metricsConsumer consumer.Metrics
+	hostMetrics     *hostMetrics
+}
+
+func newConnector(logger *zap.Logger, config component.Config) *connectorImp {
+	logger.Info("Building host_info connector")
+	cfg := config.(*Config)
+	return &connectorImp{
+		config:      *cfg,
+		logger:      logger,
+		done:        make(chan struct{}),
+		hostMetrics: newHostMetrics(),
+	}
+}
+
+// Capabilities implements connector.Traces.
+func (c *connectorImp) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// ConsumeTraces implements connector.Traces.
+func (c *connectorImp) ConsumeTraces(_ context.Context, td ptrace.Traces) error {
+	for i := 0; i < td.ResourceSpans().Len(); i++ {
+		resourceSpan := td.ResourceSpans().At(i)
+		attrs := resourceSpan.Resource().Attributes()
+		mapping := attrs.AsRaw()
+
+		for _, attrName := range c.config.HostIdentifiers {
+			if val, ok := mapping[attrName]; ok {
+				c.hostMetrics.add(val.(string))
```

Comment:
how about we add some internal telemetry, counting the number of hosts we've seen?